File: bootcamp/week-14/3-proyecto/starter/item.service.js
function createItemService(repository) {
  async function createItem(input) {
    if (!input || typeof input !== "object") {
      throw new Error("Invalid payload");
    }

    const name = typeof input.name === "string" ? input.name.trim() : "";
    const quantity = Number(input.quantity);

    if (!name) {
      throw new Error("Name is required");
    }

    if (!Number.isInteger(quantity) || quantity < 0) {
      throw new Error("Quantity must be a non-negative integer");
    }

    const exists = await repository.findByName(name.toLowerCase());
    if (exists) {
      throw new Error("Duplicated item");
    }

    return repository.save({
      name,
      quantity,
      status: quantity > 0 ? "available" : "empty",
    });
  }

  return { createItem };
}

module.exports = { createItemService };

File: bootcamp/week-16/3-proyecto/starter/item_service.py
def create_item(payload: dict) -> dict:
    name = payload.get("name", "").strip() if isinstance(payload, dict) else ""
    quantity = payload.get("quantity") if isinstance(payload, dict) else None

    if not name:
        raise ValueError("name is required")

    if not isinstance(quantity, int) or quantity < 0:
        raise ValueError("quantity must be a non-negative integer")

    return {
        "name": name,
        "quantity": quantity,
        "status": "available" if quantity > 0 else "empty",
    }
